app/jobs/geocode_firm_job.rb

```diff
@@ -0,0 +1,13 @@
+class GeocodeFirmJob < ActiveJob::Base
+  def perform(firm)
+    results = Geocoder.search(firm.full_street_address)
+
+    if results.any?
+      Stats.increment('radsignup.geocode.firm.success')
+      firm.geocode!(results.first.latitude, results.first.longitude)
+    else
+      Stats.increment('radsignup.geocode.firm.failed')
+      firm.geocode!(nil, nil)
```

Comment:
How about `Firm#geocode(lat = nil, long = nil)` so this case golfs down to `firm.geocode!`